src/GithubWebhookJira/Webhook.php

```diff
@@ -0,0 +1,276 @@
+  private $_github;
+
+  /** @var \JiraRestApi\Issue\IssueService Jira cloud issue API service */
+  private $_issue;
+
+  /** @var string The prefix for Jira issues */
+  private $_issue_prefix;
+
+  /** @var string Github Webhook Secret */
+  private $_secret = '';
+
+  /** @var string URL for Jira */
+  private $_jira_url;
+
+  /** @var string Raw data from hook request */
+  private $_raw_data = '';
+
+  /**
+   * Construct this object
+   *
+   * @param \Silex\Application $app
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   */
+  public function __construct(Application $app, Request $request) {
+    $this->_app = $app;
+    $this->_request = $request;
+
+    $this->_secret = getenv('SECRET');
+    $this->_issue_prefix = getenv('JIRA_ISSUE_PREFIX');
+    $this->_jira_url = getenv('JIRA_URL');
+
+    // Setup Github API
+    $this->_github = new Client();
+    $this->_github->authenticate(
+      getenv('GITHUB_API_TOKEN'),
+      Client::AUTH_HTTP_TOKEN
+    );
+
+    // Setup Jira API
+    $config = new ArrayConfiguration([
+      'jiraHost' => $this->_jira_url,
+      'jiraUser' => getenv('JIRA_USERNAME'),
+      'jiraPassword' => getenv('JIRA_PASSWORD'),
+      'jiraLogLevel' => 'EMERGENCY',
+    ]);
+    $this->_issue = new IssueService($config);
+
+    $this->_raw_data = $request->getContent();
+  }
+
+  /**
+   * Check if the webhook is a valid request
+   *
+   * @return boolean
+   */
+  public function isValid() : bool {
+    list($algo, $sig) = explode(
+      '=',
+      $this->_request->headers->get('X-Hub-Signature')
+    );
+    return hash_hmac($algo, $this->_raw_data, $this->_secret) === $sig;
+  }
+
+  /**
+   * Get our data
+   *
+   * @return \stdClass
+   */
+  private function _getData() : \stdClass {
+    return json_decode($this->_raw_data);
+  }
+
+  /**
+   * Process our webhook data
+   *
+   * @return void
+   */
+  public function process() {
+    switch ($this->_getData()->action) {
+      case self::ACTION_REOPENED:
+        /* falls through */
+      case self::ACTION_EDITED:
+        /* falls through */
+      case self::ACTION_OPENED:
+        $this->_processPullRequestOpen();
+        break;
+      case self::ACTION_CLOSED:
+        $this->_processPullRequestClose();
+        break;
+      default:
+        // do nothing
+    }
+  }
+
+  /**
+   * Handle opened PR action
+   */
+  private function _processPullRequestOpen() {
+    $items = $this->_getJiraItems();
+    $check = array_filter($items, function($item) {
+      return empty($item['url']);
+    });
+
+    if ( !empty($check)) {
+      $this->_updateUrls($check);
+    }
+
+    foreach ($items as $item) {
+      $transition = new Transition();
+      $transition->setTransitionId(self::TRANSITION_PEER_REVIEW);
+      $this->_issue->transition($item['key'], $transition);
+      $comment = new Comment();
+      $comment->setBody(
+        "PR Opened: [{$this->_getData()->repository->full_name}#" .
```

Comment:
Jira has good integration if the issue number is in the first line of the PR message. A comment somewhere in the conversation would be less useful. I'd like to see the Jira issue numbers appended to the end of that first line like `[ABC-123|ABC-124]` (if not already in that line)

Reply:
That might be weird if someone has a long paragraph at the beginning of their PR. I'd say it would be better to just always have the JIRA issue number put on the first line, followed by a new line and the actual PR body.

Reply:
that's what i meant. end of the first line